pkg/clustertree/cluster-manager/controllers/node_lease_controller.go

```diff
@@ -0,0 +1,212 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	coordinationv1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kosmos.io/kosmos/pkg/utils"
+)
+
+const (
+	NodeLeaseControllerName = "node-lease-controller"
+
+	DefaultLeaseDuration         = 40
+	DefaultRenewIntervalFraction = 0.25
+
+	DefaultNodeStatusUpdateInterval = 1 * time.Minute
+)
+
+type NodeLeaseController struct {
+	leafClient kubernetes.Interface
+	rootClient kubernetes.Interface
+	root       client.Client
+
+	leaseInterval  time.Duration
+	statusInterval time.Duration
+
+	node     *corev1.Node
+	nodeLock sync.Mutex
+}
+
+func NewNodeLeaseController(leafClient kubernetes.Interface, root client.Client, node *corev1.Node, rootClient kubernetes.Interface) *NodeLeaseController {
+	c := &NodeLeaseController{
+		leafClient:     leafClient,
+		rootClient:     rootClient,
+		root:           root,
+		node:           node,
+		leaseInterval:  getRenewInterval(),
+		statusInterval: DefaultNodeStatusUpdateInterval,
+	}
+	return c
+}
+
+func (c *NodeLeaseController) Start(ctx context.Context) error {
+	c.syncLease(ctx)
+	go wait.UntilWithContext(ctx, c.syncLease, c.leaseInterval)
+
+	c.syncNodeStatus(ctx)
```

Comment:
it is not necessary to  invoke function `syncNodeStatus`  twice.

Reply:
Is there a good way to do this? I want to execute it immediately.

Reply:
Oh I see, I was wrong about the UntilWithContext

Reply:
UntilWithContext will execute right now in the first time

Reply:
The difference between `UntilWithContext` and `NonSlidingUntilWithContext` .